packages/adapter-squid/src/SquidAdapter.ts

```diff
@@ -0,0 +1,226 @@
+      private adapterConfigValidator: IValidator<SquidAdapterConfig>,
+      @inject(StepConfigValidator)
+      private stepConfigValidator: IValidator<SquidStepConfig>,
+      @inject(SquidProvider) private squidProvider: ISquidProvider,
+      @inject(ConfigProvider) private configProvider: IConfigProvider,
+      @inject(SignerService) private signerService: ISignerService
+   ) {}
+
+   public async init(): Promise<Boolean> {
+      if (this.adapterConfig) {
+         return true;
+      }
+
+      const adapterConfigItem = this.configProvider.getAdapterConfig(
+         this.adapterId
+      );
+
+      if (!adapterConfigItem) {
+         throw new Error(
+            `${ERR_ADAPTER_INIT_FAILURE}: ${ERR_ADAPTER_CONFIG_MISSING}`
+         );
+      }
+
+      try {
+         this.adapterConfig = await this.adapterConfigValidator.validate(
+            adapterConfigItem.config
+         );
+
+         const squidConfig = {
+            baseUrl: this.adapterConfig.baseUrl,
+            integratorId: this.adapterConfig.integratorId,
+         };
+
+         await this.squidProvider.init(squidConfig);
+      } catch (error: any) {
+         throw new Error(
+            `${ERR_ADAPTER_INIT_FAILURE}: ${(error as Error).message}`
+         );
+      }
+
+      return true;
+   }
+
+   public async isValid(
+      step: Step<SquidStepConfig>
+   ): Promise<ValidationResult> {
+      let result: ValidationResult = {
+         isValid: false,
+         errors: [],
+      };
+
+      try {
+         await this.stepConfigValidator.validate(step.config);
+         result.isValid = true;
+      } catch (err) {
+         const validationErr = err as ValidationError;
+
+         if (validationErr.message) {
+            result.errors.push(validationErr.message);
+         }
+         if (validationErr.context) {
+            result.errors.push(validationErr.context);
+         }
+      }
+
+      return result;
+   }
+   // TODO: Get some logging set up.
+
+   public async execute(step: Step<SquidStepConfig>): Promise<ExecutionResult> {
+      // We're assuming the step is Bridge.Swap as this is the only step type currently supported
+      // Should this change, this function should be refactored to handle the different step types.
+      // e.g. IExecutor.executeStep(step: Step<SquidStep>)
+
+      const result: ExecutionResult = {
+         success: false,
+         data: {},
+      };
+
+      const squid = this.squidProvider.getSquid();
+      const signer = this.signerService.getSignerForChain(
+         step.config.fromChain
+      );
+
+      let transaction;
+
+      try {
+         const route = await this.getRoute(step.config, squid);
+         transaction = await squid.executeRoute({ signer, route });
+      } catch (error) {
+         result.success = false;
+         result.data.errorMessage = `${ERR_ADAPTER_EXECUTE_FAILURE}:${
+            (error as Error).message
+         }`;
+         return result;
+      }
```

Comment:
why `let` over `const`? AFAICS we're not reassigning `result` anywhere


```suggestion
    const result: ValidationResult = {
```